nuxeo-features/preview/nuxeo-preview-core/src/main/java/org/nuxeo/ecm/platform/preview/helper/PreviewHelper.java

```diff
@@ -49,7 +52,12 @@ public static String getPreviewURL(DocumentModel doc, String xpath) {
         if (xpath != null) {
             sj.add("@blob").add(xpath);
         }
-        return sj.add("@preview").toString();
+        String result = sj.add("@preview").toString();
+        String ct = doc.getChangeToken();
+        if (StringUtils.isNotBlank(ct)) {
+            result += "?" + CoreSession.CHANGE_TOKEN + "=" + StringEscapeUtils.escapeHtml4(ct);
```

Comment:
That's not the correct escaping method. Here you want `URLEncoder.encode(ct, "UTF-8")`